flowmachine/flowmachine/features/subscriber/scores.py

```diff
@@ -112,20 +102,73 @@ class EventScore(Query):
 
     def __init__(
         self,
-        start,
-        stop,
-        level="admin3",
-        hours="all",
-        table="all",
-        score_hour={(7, 9): 0, (9, 16): 1, (16, 20): 0, (20, 7): -1},
-        score_dow={(1, 5): 1, (5, 5): 0, (6, 1): -1},
-        subscriber_identifier="msisdn",
-        column_name=None,
         *,
+        start: str,
+        stop: str,
+        level: str = "admin3",
+        hours: str = "all",
+        table: str = "all",
+        score_hour: List[float] = [
+            -1,
+            -1,
+            -1,
+            -1,
+            -1,
+            -1,
+            -1,
+            0,
+            0,
+            1,
+            1,
+            1,
+            1,
+            1,
+            1,
+            1,
+            1,
+            0,
+            0,
+            0,
+            0,
+            -1,
+            -1,
+            -1,
+        ],
+        score_dow: Dict[str, float] = {
+            "monday": 1,
+            "tuesday": 1,
+            "wednesday": 1,
+            "thursday": 0,
+            "friday": -1,
+            "saturday": -1,
+            "sunday": -1,
+        },
+        subscriber_identifier: str = "msisdn",
+        column_name: Union[str, List[str]] = None,
         subscriber_subset=None,
-        **kwargs,
+        polygon_table=None,
+        size=None,
     ):
-
+        if set(score_dow.keys()) != {
+            "monday",
+            "tuesday",
+            "wednesday",
+            "thursday",
+            "friday",
+            "saturday",
+            "sunday",
+        }:
+            raise ValueError(
+                f"Day of week score dictionary must have values for all days. Only got {set(score_dow.keys())}"
+            )
+        if len(score_hour) != 24:
+            raise ValueError(
+                f"Hour of day score dictionary must have values for all hours. Only got {len(score_hour)}"
```

Comment:
Maybe remove "only" because implies that the user always provides fewer than 24 hours?